Add UI information about AAQ operator (#111679)
openshift/openshift-docs: modules/virt-creating-vm-quotas-web.adoc

// Module included in the following assemblies:
//
// * virt/managing_vms/advanced_vm_management/virt-working-with-resource-quotas-for-vms.adoc

:_mod-docs-content-type: PROCEDURE
[id="virt-creating-vm-quotas-web_{context}"]
= Creating virtual machine quotas using the web console

[role="_abstract"]

> You can create Application Aware Quota (AAQ) quotas that limit virtual machine resource consumption by using the guided form in the {product-titles} web console. The form is the recommended approach for simple virtualization quota use cases.

Reviewer: ```suggestion
You can create Application Aware Quota (AAQ) quotas that limit virtual machine resource consumption by using the guided form in the {product-title} web console. The form is the recommended approach for simple virtualization quota use cases.
```

.Prerequisites

> * You have administrator privileges in the OpenShift cluster.

Reviewer: Optional, for consistency
```suggestion
* You have cluster administrator privileges.
```

* The Application Aware Quota operator is enabled in your cluster. See xref:virt-enabling-aaq-operator_{context}[Enabling the Application Aware Quota operator].
* The project where you want to create the quota exists.

.Procedure

. In the {product-title} web console, click *Virtualization* -> *Quotas*.

. Click *Create quota*.

. In the dropdown menu, select *With form*.
+
The create quota form is displayed.

. Enter a name for your quota in the *Name* field.
+
The name must be unique within the project.

. Select the project where you want to apply this quota.
+
By default, the current project is selected. You can click the dropdown to select a different project if needed.
+
[NOTE]
====
If a Kubernetes `ResourceQuota` already exists in the selected project, an alert is displayed with a link to view the existing ResourceQuota. You can still create the AAQ quota, but both quotas will apply to the project.
====

> . Configure CPU, memory, and VMI limits for the quota:

Author: @adamviktora Let me know if this step looks better, and if there's a way I can improve it

Reviewer 2: This is great! Thank you @rheslop

+

> The fields displayed vary based on the selected quota calculation method:

Reviewer 3: Just for awareness, there are other configuration types as well, but I assume this can work as a starting point.

> The fields displayed vary based on the selected quota calculation method:

Author: ```suggestion
The fields displayed vary based on the selected quota calculation method. For example:
```
Should I expand the section below to include additional quota calculation methods?

+
* For the  `Virtual resources` or `Dedicated virtual resources` quota calculation method, the quotas exclude pod overhead:

> ** Use the *vCPU allocation* field to set the maximum number of virtual CPUS permitted for all VMs in the project, excluding infrastructure overhead.

Reviewer: ```suggestion
** Use the *vCPU allocation* field to set the maximum number of virtual CPUs permitted for all VMs in the project, excluding infrastructure overhead.
```

> ** Use the *Virtual memory allocattion* field to set the maximum memory capicity, in GiB, excluding pod overhead. 

Reviewer 3: ```suggestion
** Use the *vCPU allocation* field to set the maximum number of virtual CPUS permitted for all VMs in the project, excluding infrastructure overhead.
** Use the *Virtual memory allocattion* field to set the maximum memory capicity, in GiB, excluding infrastructure overhead. 
```

> ** Use the *Virtual memory allocattion* field to set the maximum memory capicity, in GiB, excluding infrastructure overhead. 

Reviewer: ```suggestion
** Use the *Virtual memory allocation* field to set the maximum memory capacity, in GiB, excluding infrastructure overhead. 
```

** Use the *VMI limits* field to set the maximum number of virtual machine instances.

> * For the `Virtual Maching Instance (VMI) pod usage` quota calculation method, the quotas include pod overhead:

Reviewer 3: Although virtualization overhead is included in this configuration, live migration overhead is still excluded.

> * For the `Virtual Maching Instance (VMI) pod usage` quota calculation method, the quotas include pod overhead:

Reviewer: ```suggestion
* For the `Virtual Machine Instance (VMI) pod usage` quota calculation method, the quotas include pod overhead:
```

> ** Use the *CPU allocation* field to set the maximum number of CPU cores for pods that run VM workloadss

Reviewer: ```suggestion
** Use the *CPU allocation* field to set the maximum number of CPU cores for pods that run VM workloads
```

> ** Use the *Memory allocation* field to set the maximum memory capicity, in GiB, for pods that run VM workloads.

Reviewer: ```suggestion
** Use the *Memory allocation* field to set the maximum memory capacity, in GiB, for pods that run VM workloads.
```

> ** Use the *VMI limits* field to set the maximum number of virtual machine instances.

Author: ```suggestion
** Use the *VMI limits* field to set the maximum number of virtual machine instances.
+
[NOTE]
====
Live migration overhead is excluded from the `Virtual Machine Instance (VMI) pod usage`  quota calculation method.
====
```
How is this?

. Review the configuration and click *Create*.
+
The quota is created and you are redirected to the quota details page. The new quota is now active and enforcing resource limits on virtual machines in the selected project.

.Verification

> . In the OpenShift web console, click *Virtualization* -> *Quotas*.

Reviewer: ```suggestion
. In the {product-title} web console, click *Virtualization* -> *Quotas*.
```

// Module included in the following assemblies:
//
// * virt/managing_vms/advanced_vm_management/virt-working-with-resource-quotas-for-vms.adoc

:_mod-docs-content-type: PROCEDURE
[id="virt-creating-vm-quotas-web_{context}"]
= Creating virtual machine quotas using the web console

[role="_abstract"]
You can create Application Aware Quota (AAQ) quotas that limit virtual machine resource consumption by using the guided form in the {product-titles} web console. The form is the recommended approach for simple virtualization quota use cases.

.Prerequisites

* You have administrator privileges in the OpenShift cluster.
* The Application Aware Quota operator is enabled in your cluster. See xref:virt-enabling-aaq-operator_{context}[Enabling the Application Aware Quota operator].
* The project where you want to create the quota exists.

.Procedure

. In the {product-title} web console, click *Virtualization* -> *Quotas*.

. Click *Create quota*.

> . In the dropdown menu, select *Create VM-specific quota*.

Reviewer 2: The dropdown options are called "With form" and "With YAML"

Author: ```suggestion
. In the dropdown menu, select *With form*.
```

+
The create quota form is displayed.

> . Enter a name for your quota in the *Quota name* field.

Reviewer 2: The field label is just "Name"

Author: ```suggestion
. Enter a name for your quota in the *Name* field.
```

+
The name must be unique within the project.

. Select the project where you want to apply this quota.
+
By default, the current project is selected. You can click the dropdown to select a different project if needed.
+
[NOTE]
====
If a Kubernetes `ResourceQuota` already exists in the selected project, an alert is displayed with a link to view the existing ResourceQuota. You can still create the AAQ quota, but both quotas will apply to the project.
====

. Optional: Configure the vCPU allocation for virtual machines in this project.
+
In the *vCPU allocation* field, enter the maximum number of virtual CPUs permitted for all combined VMs in this project. This measurement excludes pod runtime resource consumption.

. Optional: Configure the virtual memory allocation for virtual machines in this project.
+

> In the *Virtual memory allocation* field, enter the maximum memory capacity in GiB (gigabytes) permitted for all combined VMs in this project. This measurement excludes pod runtime resource consumption.

Reviewer 2: `GiB` stands for `gibibytes` to be precise (1024 ^ 3 bytes), gigabytes are decimal based (1000 ^ 3)

Author: ```suggestion
In the *Virtual memory allocation* field, enter the maximum memory capacity in GiB (gibibytes) permitted for all combined VMs in this project. This measurement excludes pod runtime resource consumption.
```

. Optional: Configure CPU allocation for VM pods.
+

> In the *CPU allocation* field, enter the maximum CPU cores permitted for pods that run VM workloads. This includes both the VM and pod runtime resource consumption.

Reviewer 2: just a note that this field is mutually exclusive with "vCPU allocation" - which one is displayed is based on the selected Quota calculation method (same for Virtual memory / Memory allocation)

Author: I want to work something in to the procedure to describe this, but I'm having trouble finding the information on the design proposal (https://docs.google.com/document/d/1WMoQlhR2izADE7sPlG6p82oOapWeQV3krOCrJJ9RLu8/edit?tab=t.827hp54k2bt7)

I see the screen shot change from vCPU allocation / Virtual memory allocation to vCPU limits / Memory limits when showing a tool tip, (so I'm not sure my labels are correct) but I also don't see explanatory text.

Can you help?

Reviewer 2: It looks like the design might be a little bit outdated. This is a video of the current implementation:

https://github.com/user-attachments/assets/1a8dabb7-b0ac-4d8d-8ea6-391feb562d1a

Hope it helps, let me know if you have any questions

* The project where you want to create the quota exists.

.Procedure

. In the {product-title} web console, click *Virtualization* -> *Quotas*.

. Click *Create quota*.

. In the dropdown menu, select *With form*.
+
The create quota form is displayed.

. Enter a name for your quota in the *Name* field.
+
The name must be unique within the project.

. Select the project where you want to apply this quota.
+
By default, the current project is selected. You can click the dropdown to select a different project if needed.
+
[NOTE]
====
If a Kubernetes `ResourceQuota` already exists in the selected project, an alert is displayed with a link to view the existing ResourceQuota. You can still create the AAQ quota, but both quotas will apply to the project.
====

. Configure CPU, memory, and VMI limits for the quota:
+
The fields displayed vary based on the selected quota calculation method. For example:
+
* For the  `Virtual resources` or `Dedicated virtual resources` quota calculation method, the quotas exclude pod overhead:
** Use the *vCPU allocation* field to set the maximum number of virtual CPUs permitted for all VMs in the project, excluding infrastructure overhead.

> ** Use the *Virtual memory allocation* field to set the maximum memory capacity, in GiB, excluding infrastructure overhead. 

Reviewer 3: Shouldn't we also explain the difference between the two configurations? With Dedicated Virtual Resources, VM CPU and memory resources are counted against dedicated VM-specific quota limits. In contrast, Virtual Resources aggregate VM resource usage into the standard built-in CPU and memory request/limit quotas that are shared across all pod workloads.

Author: These differences are explained in "Enabling the Application Aware Quota Operator" 👍

** Use the *VMI limits* field to set the maximum number of virtual machine instances.

> * For the `Virtual Machine Instance (VMI) pod usage` quota calculation method, the quotas include pod overhead:

Reviewer 3: ```suggestion
* For the `Virtual Machine Instance (VMI) pod usage` quota calculation method, quotas are calculated based on the actual resources consumed by the VMI pod, including virtualization and infrastructure overhead, but excluding the additional resources temporarily required during live migration:
```

Reviewer 3: Oh i just noticed we already include this info in the NOTE below please ignore this comment
